test/testcafe/spec/IdentifyRecoveryWithCaptcha_spec.js

```diff
@@ -0,0 +1,95 @@
+import { RequestMock, RequestLogger, Selector } from 'testcafe';
+import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
+import { checkConsoleMessages } from '../framework/shared';
+import xhrIdentifyRecoveryWithRecaptcha from '../../../playground/mocks/data/idp/idx/identify-recovery-with-recaptcha-v2.json';
+import xhrIdentifyRecoveryWithHcaptcha from '../../../playground/mocks/data/idp/idx/identify-recovery-with-hcaptcha.json';
+import xhrAuthenticatorVerifySelect from '../../../playground/mocks/data/idp/idx/authenticator-verification-select-authenticator';
+
+const identifyRecoveryWithReCaptchaMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecoveryWithRecaptcha)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRecoveryWithHCaptchaMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecoveryWithHcaptcha)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRequestLogger = RequestLogger(
+  /idx\/identify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+fixture('Identify Recovery - reset flow with Captcha');
+
+async function setup(t) {
+  const identityPage = new IdentityPageObject(t);
+  await identityPage.navigateToPage();
+  await checkConsoleMessages({
+    controller: 'forgot-password',
+    formName: 'identify-recovery',
+  });
+
+  return identityPage;
+}
+
+// https://oktainc.atlassian.net/browse/OKTA-393059
+// We're disabling this test for now because there seems to be an underlying issue with with this test
+// in Bacon. Locally this test runs with no issues but it's very flaky on Bacon. OKTA-393059 is created to investigate
+// further.
+test.requestHooks(identifyRequestLogger, identifyRecoveryWithReCaptchaMock).skip('should be able to submit identifier with reCaptcha enabled', async t => {
```

Comment:
We skip this test (and the reCaptcha ones in other files) for now because there's additional investigation required to know why the reCaptcha eng-leaks and testCafe tests fail on Bacon only (https://oktainc.atlassian.net/browse/OKTA-393059). It's currently not clear as to why they can't pass clicking the submit button. In the test we already assert that the Captcha related DOM loads correctly but despite this, we can't get past the submit button sometimes.

Reply:
UPDATE: the test has been re-written in a different way to avoid the limitation we see in Bacon regarding reCaptcha lib. Now during the reCaptcha tests, we just validate whether we hit google's API with the expected siteKey during submission. From my troubleshooting, this is the best option we have as it's otherwise hard to mock the reCaptcha service and the overall flow. For our purposes this will suffice for validating that the functionality is working as expected.